### PR TITLE
AP-2226: Merits task list

### PR DESCRIPTION
**Pull request by colinbruce**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/AP-2226)

Update handling of the Multi Proceeding merits task list, this allows tasks to be completed upon exiting a page and
prevents a user from continuing past the task list without completing all required steps.

*NOTE* I would recommend reviewing by commit rather than as a whole.  I tried to group relevant steps together to ease the cognitive load of reviewing all 62 files at once! 😄 

## Checklist

Before you ask people to review this PR:

- [x] Tests and rubocop should be passing: `bundle exec rake`
- [x] Github should not be reporting conflicts; you should have recently run `git rebase master`.
- [x] There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- [x] The PR description should say what you changed and why, with a link to the JIRA story.
- [x] You should have looked at the diff against master and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: spec/requests/providers/application_merits_task/statement_of_case_spec.rb [hunk 1]
```diff
@@ -65,18 +71,60 @@ module ApplicationMeritsTask
...
+
+              context 'and no involved children exist' do
+                it 'redirects to the next page' do
+                  subject
+                  expect(response).to redirect_to new_providers_legal_aid_application_involved_child_path(legal_aid_application)
+                end
+              end
+            end
+
+            # context 'and the application does not have a section 8 proceeding type' do
```
Comment (willc-work) [inline spec/requests/providers/application_merits_task/statement_of_case_spec.rb:114]: can this be removed now?

File: app/views/providers/merits_task_lists/show.html.erb [hunk 2]
```diff
@@ -1,22 +1,31 @@
...
+  <div id='task_list' class="<%= 'govuk-form-group--error' if @merits_task_list&.errors&.any? && @merits_task_list.errors[:task_list].present? %>">
+    <h2 class="govuk-heading-l"><%= t('.application_heading') %></h2>
 
-    <h2 class="govuk-heading-l govuk-!-margin-top-4"><%= t('.proceeding_heading') %></h2>
-    <% @merits_tasks[:proceedings].each_with_index do |(_, proceeding), index| %>
-      <%= render 'task_list_section', name: proceeding[:name], tasks: proceeding[:tasks], ccms_code: @merits_tasks[:proceedings].keys[index], section_number: index + 2 %>
+    <% if @merits_task_list&.errors&.any? && @merits_task_list.errors[:task_list].present? %>
+      <span class="govuk-error-message" id="task-list-error">
+        <span class="govuk-visually-hidden">Error: </span>
+        <%= @merits_task_list&.errors[:task_list]&.first %>
```
Comment (willc-work) [inline app/views/providers/merits_task_lists/show.html.erb:13]: are these safe operators required here? I dont think code wont be executed if @merits_task_list.errors[:task_list].first

**Review by willc-work — APPROVED** [review]

**Review by willc-work — APPROVED** [review]
Looks great, 2 queries nothing major.

File: app/views/providers/merits_task_lists/show.html.erb (hunk 2 above)
Comment (CeeBeeUK) [reply app/views/providers/merits_task_lists/show.html.erb:13]: I think I added them for a reason, but that may have changed later. I'll try removing it and see what happens! 

I don't want this blocking stuff much longer 👍

Comment (willc-work) [reply app/views/providers/merits_task_lists/show.html.erb:13]: In that case I dont feel strongly enough about them that they have to be removed

Comment (colinbruce) [reply app/views/providers/merits_task_lists/show.html.erb:13]: Double checked and you were right.  I think the model, originally, couldn't handle errors so it was a separate error model for a while until I refactored it back into the form. :+1:

**Pull request merged**